Limit line.Split in ConfigFile.Reload()

xoxfaby opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->


## Description
<!--- Describe your changes in detail -->
Limiting the line.Split in Reload() to 2 substrings to allow for values containing `=` characters. Original functionality is maintained.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
I stumbled across this bug when trying to use `=` characters in my config values.  
I see no reason config values shouldn't be allowed to contain `=`'s, I assume this was an oversight when originally implemented. 

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how -->
<!--- your change affects other areas of the code, etc. -->

This is such a small, basic change that I have not tested it but if absolutely necessary I can test it.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

denikson commented on BepInEx/Configuration/ConfigFile.cs:
```diff
@@ -118,7 +118,7 @@ public void Reload()
 					}
 
-					string[] split = line.Split('='); //actual config line
+					string[] split = line.Split('=',2); //actual config line
```
.NET Framework 3.5 does not have an overload `String.Split(char, int)`, so this will not compile. If you go this way, you'd have to pick [an overload that's available in FW 3.5](https://docs.microsoft.com/en-us/dotnet/api/system.string.split?view=netframework-3.5) as that's the target BepInEx uses.

denikson reviewed (changes requested):
Greetings! 

Thank you for the pull request. I've added an important note regarding the change, as it will not compile now.

However, there is a bigger issue at hand: what if having `=` is expected input for a value? In my opinion, a better option is to pick the first `=` and include the rest to be part of the value (which is doable with a `IndexOf` and a `Substring`). While still not perfect, it would be a good enough workaround until 6.0 where we are intending on using full TOML parser.

xoxfaby commented:
Oh I see, my bad.

As for the other comment, if String.Split is limited, the last substring will contain the rest of the string, that was the intention with this change. 

`String.Split(['=`], 2)` should compile and work as you described.

denikson commented:
> As for the other comment, if String.Split is limited, the last substring will contain the rest of the string, that was the intention with this change.

Ah, that is indeed true. In that case just a split should work.

> `String.Split(['='], 2)` should compile and work as you described.

That's still not valid C#; to define an array you would use `new []{ '=' }`.

Would you mind pushing the change so I can merge the PR or are you fine with me making a manual push to master directly?

xoxfaby commented:
Derp, thank you, my brain is still in python land sometimes. Now it should be all good?

denikson commented:
Yes, looks good to me!
Thank you for the PR again!
